### PR TITLE
refactor(disk-fill): Converting the daemonset into job

### PR DESCRIPTION
**Pull request by rajdas98**
Signed-off-by: Raj <raj.das@mayadata.io>
- Converting the daemonset to job

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Checklist**

* [x] Does this PR have a corresponding GitHub issue?
https://github.com/litmuschaos/litmus/issues/1265
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by rajdas98** [comment]
[logs.txt](https://github.com/litmuschaos/litmus/files/4298683/logs.txt)

File: chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml [hunk 1]
```diff
@@ -61,17 +61,20 @@
...
     - name: Patch the engine UID
       template:
-        src:  /chaoslib/litmus/disk_fill/disk_fill_ds.j2
-        dest: /chaoslib/litmus/disk_fill/disk_fill_ds.yml
-      
-    - name: Create the DaemonSet for Disk-Fill
+        src:  /chaoslib/litmus/disk_fill/disk_fill_job.j2
+        dest: /chaoslib/litmus/disk_fill/disk_fill_job.yml
+      vars:
+        chaosUID: "{{ chaos_uid }}"
```
Comment (ispeakc0de) [inline chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml]: Can we directly use `chaos_uid` variable inside jinja template as it is globally available? pls refer any other experiments

File: chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml [hunk 2]
```diff
@@ -61,17 +61,20 @@
...
+        app_node: "{{ app_node }}"
```
Comment (ispeakc0de) [inline chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml]: If the name of both variables are the same why are you passing this? Why can you directly use the `app_node`  variable there without passing it?

File: chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml [hunk 3]
```diff
@@ -61,17 +61,20 @@
...
+        kubectl apply -f /chaoslib/litmus/disk_fill/disk_fill_job.yml -n {{ a_ns }}
       args:
         executable: /bin/bash
-      register: disk_fill_ds_result
+      register: disk_fill_job_result
   
-    - name: Confirm that the disk-fill daemonSet is running on all nodes
+    - name: Confirm that the disk-fill job is running on all nodes
       shell: >
         kubectl get pods -l app=disk-fill --no-headers --template "{{range .items}}{{if eq .spec.nodeName $(kubectl get nodes --no-headers --field-selector spec.unschedulable=false | grep Ready | awk '{printf "\"%s\" ", $1}')}}{{.status.phase}}{{\"\n\"}}{{end}}{{end}}" -n {{ a_ns }} | sort | uniq
```
Comment (ispeakc0de) [inline chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml]: We are you adding a complex task to check the status of job. Can you please simply it?

File: chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml [hunk 4]
```diff
@@ -61,17 +61,20 @@
...
       shell: >
-        kubectl apply -f /chaoslib/litmus/disk_fill/disk_fill_ds.yml -n {{ a_ns }}
+        kubectl apply -f /chaoslib/litmus/disk_fill/disk_fill_job.yml -n {{ a_ns }}
       args:
         executable: /bin/bash
-      register: disk_fill_ds_result
+      register: disk_fill_job_result
   
-    - name: Confirm that the disk-fill daemonSet is running on all nodes
+    - name: Confirm that the disk-fill job is running on all nodes
```
Comment (ispeakc0de) [inline chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml]: How this job is running on all node?

File: chaoslib/litmus/disk_fill/disk_fill_job.j2 [hunk 5]
```diff
@@ -1,21 +1,19 @@
...
-    matchLabels:
-      app: disk-fill
-      chaosUID: {{ chaos_uid }}
   template:
     metadata:
       labels:
         app: disk-fill
-        chaosUID: {{ chaos_uid }}
-      name: disk-fill
+        chaosUID: {{ chaosUID }}
```
Comment (ispeakc0de) [inline chaoslib/litmus/disk_fill/disk_fill_job.j2]: chaosUID: {{ chaos_uid }}

**Comment by ispeakc0de** [comment]
> [logs.txt](https://github.com/litmuschaos/litmus/files/4298683/logs.txt)

Can you please provide the update logs

**Comment by rajdas98** [comment]
[logs.txt](https://github.com/litmuschaos/litmus/files/4305434/logs.txt)

File: chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml [hunk 6]
```diff
@@ -59,11 +59,11 @@
         executable: /bin/bash
       register: app_node
 
-    - name: Patch the engine UID
+    - name: Create the Job for Disk-Fill
```
Comment (ispeakc0de) [inline chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml]: Here we just patching engine_uid, node name, etc so the name should be relevant

File: chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml [hunk 7]
```diff
@@ -73,23 +73,24 @@
...
       shell: >
-        kubectl apply -f /chaoslib/litmus/disk_fill/disk_fill_ds.yml -n {{ a_ns }}
+        kubectl apply -f /chaoslib/litmus/disk_fill/disk_fill_job.yml -n {{ a_ns }}
       args:
         executable: /bin/bash
-      register: disk_fill_ds_result
+      register: disk_fill_job_result
   
-    - name: Confirm that the disk-fill daemonSet is running on all nodes
+    - name: Confirm that the disk-fill job is running on {{ app_node }} node
```
Comment (ispeakc0de) [inline chaoslib/litmus/disk_fill/disk_fill_by_litmus.yml]: This and "Confirm that disk fill container is in running state " task both are doing the same thing so can you please remove this task?

**Pull request merged**